airflow/contrib/operators/cloudml_operator.py

```diff
@@ -406,3 +411,136 @@ def execute(self, context):
                                        self._version['name'])
         else:
             raise ValueError('Unknown operation: {}'.format(self._operation))
+
+
+class CloudMLTrainingOperator(BaseOperator):
+    """Operator for launching a CloudML training job.
+
+    :param project_name: The Google Cloud project name within which CloudML
+        training job should run.
+    :type project_name: string
+
+    :param job_id: A unique templated id for the submitted Google CloudML
+        training job.
+    :type job_id: string
+
+    :param package_uris: A list of package locations for CloudML training job,
+        which should include the main training program + any additional
+        dependencies.
+    :type package_uris: string
+
+    :param training_python_module: The Python module name to run within CloudML
+        training job after installing 'package_uris' packages.
+    :type training_python_module: string
+
+    :param training_args: A list of templated command line arguments to pass to
+        the CloudML training program.
+    :type training_args: string
+
+    :param region: The Google Compute Engine region to run the CloudML training
+        job in.
+    :type region: string
+
+    :param scale_tier: Resource tier for CloudML training job.
+    :type scale_tier: string
+
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: string
+
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: string
+
+    :param mode: Can be one of 'DRY_RUN'/'CLOUD'. In 'DRY_RUN' mode, no real
+        training job will be launched, but the CloudML training job request
+        will be printed out. In 'CLOUD' mode, A real CloudML training job
+        creation request will be issued.
+    :type mode: string
+    """
+
+    template_fields = [
+        '_job_id',
```

Comment:
Does it make sense to enable templating for other variables, such as project_name? Reason I ask is that a templated project_name would allow for project_name variable templating to be done in Airflow (i.e. you could have a variable in airflow called 'project', and reference that as a templated variable when creating the operator).